prow/config/config.go

```diff
@@ -1295,6 +1302,52 @@ func resolvePresets(name string, labels map[string]string, spec *v1.PodSpec, bui
 	return nil
 }
 
+var ReProwExtraRef = regexp.MustCompile(`PROW_EXTRA_GIT_REF_(\d+)`)
+
+func ValidatePipelineRunSpec(jobType prowapi.ProwJobType, extraRefs []prowapi.Refs, spec *pipelinev1alpha1.PipelineRunSpec) error {
+	if spec == nil {
+		return nil
+	}
+	// Validate that that the refs match what is requested by the job.
+	// The implicit git ref is optional to use, but any extra refs specified must
+	// be used or removed. (Specifying an unused extra ref must always be
+	// unintentional so we want to warn the user.)
+	extraIndexes := sets.NewInt()
+	for _, resource := range spec.Resources {
+		// Validate that periodic jobs don't request an implicit git ref
+		if jobType == prowapi.PeriodicJob && resource.ResourceRef.Name == ProwImplicitGitResource {
+			return fmt.Errorf("periodic jobs do not have an implicit git ref to replace %s", ProwImplicitGitResource)
+		}
+
+		match := ReProwExtraRef.FindStringSubmatch(resource.ResourceRef.Name)
+		if len(match) != 2 {
+			continue
+		}
+		if len(match[1]) > 1 && match[1] == "0" {
```

Comment:
I'm not sure this is correct, we check `match[1]` is greater in length than 1 character and then check if `match[1] == "0"`, a single `0`?

I had a play in the playground and can pass `PROW_EXTRA_GIT_REF_001` through this code without being invalidated, I think we are trying to catch that?
https://play.golang.org/p/U3Tm2v6dGkc

Reply:
Good catch! I meant for this to check the first character, not the entire string.
I fixed this and added tests for this function.